overlord/configstate/configcore/netplan.go

```diff
@@ -49,22 +57,44 @@ func init() {
 	config.RegisterExternalConfig("core", "system.network.netplan", getNetplanFromSystem)
 }
 
-func validateNetplanSettings(tr config.Conf) error {
-	// XXX: validate somehow once we support writing?
-	return nil
+type connectivityCheckStore interface {
+	ConnectivityCheck() (map[string]bool, error)
 }
 
-func handleNetplanConfiguration(tr config.Conf, opts *fsOnlyContext) error {
-	for _, chg := range tr.Changes() {
-		if chg == "core.system.network.netplan" || strings.HasPrefix(chg, "core.system.network.netplan.") {
-			return fmt.Errorf("cannot set netplan config yet")
+var snapstateStore = func(st *state.State, deviceCtx snapstate.DeviceContext) connectivityCheckStore {
+	return snapstate.Store(st, deviceCtx)
+}
```

Comment:
Any reason why this is not just `var snapstateStore = snapstate.Store`?

Reply:
Probably not, silly me! I can look into this next.

Reply:
Aha, actually I remember now - I think the reaosn is that I wanted to return the connectivityCheckStore interface only so that I need to mock less :)